contracts/PowerPass.sol

```diff
@@ -0,0 +1,99 @@
+
+contract PowerPass is Initializable, Ownable, ERC721 {
+    using LibString for uint256;
+    /*//////////////////////////////////////////////////////////////
+                                STORAGE
+    //////////////////////////////////////////////////////////////*/
+    address public claimSignerAddress;
+    uint256 private _tokenIdCounter;
+
+    /*//////////////////////////////////////////////////////////////
+                                 ERRORS
+    //////////////////////////////////////////////////////////////*/
+    error AddressNotSigned();
+    error TokenNotTransferable();
+    error AddressAlreadyMinted();
+
+    /*//////////////////////////////////////////////////////////////
+                              CONSTRUCTOR
+    //////////////////////////////////////////////////////////////*/
+    /// @custom:oz-upgrades-unsafe-allow constructor
+    // solhint-disable-next-line func-visibility
+    constructor() {
+        _disableInitializers();
+    }
+
+    function initialize(
+        address owner_,
+        address claimSignerAddress_
+    ) external initializer {
+        _initializeOwner(owner_);
+        claimSignerAddress = claimSignerAddress_;
+    }
+
+    /*//////////////////////////////////////////////////////////////
+                            EXTERNAL UPDATE
+    //////////////////////////////////////////////////////////////*/
+    /// @dev universal dynamic mint function
+    /// @param signature_ The signature of the data
+    /// @param data_ The data to decode for the claim
+    function mint(bytes calldata signature_, bytes calldata data_) external payable {
+        address to_ = abi.decode(data_, (address));
+        if (recoverSigner(keccak256(data_), signature_) != claimSignerAddress) revert AddressNotSigned();
+        if (balanceOf(to_) > 0) revert AddressAlreadyMinted();
+
+        _tokenIdCounter += 1;
+        _mint(to_, _tokenIdCounter);
+    }
+
+    /// @dev set the claim signer address
+    /// @param claimSignerAddress_ The address of the claim signer
+    function setClaimSignerAddress(address claimSignerAddress_) external onlyOwner {
+        claimSignerAddress = claimSignerAddress_;
+    }
+
+    /*//////////////////////////////////////////////////////////////
+                             EXTERNAL VIEW
+    //////////////////////////////////////////////////////////////*/
+
+    /// @dev Returns the token collection name.
+    function name() public pure override returns (string memory){
+        return "RabbitHole Power Pass";
+    }
+
+    /// @dev Returns the token collection symbol.
+    function symbol() public pure override returns (string memory){
+        return "RHPP";
+    }
+
+    /// @dev Returns the Uniform Resource Identifier (URI) for token `id`.
+    function tokenURI(uint256 id) public pure override returns (string memory){
+        return LibString.concat("https://api.rabbithole.gg/v1/powerpass/", id.toString());
```

Comment:
Are you planning to deploy this behind a proxy? Might make sense to make the URI settable if we arn't going to have this upgradable.